search/src/Components/upload/react-images-upload/index.js

```diff
@@ -85,55 +87,47 @@ class ReactImageUploadComponent extends React.Component {
           type: ERROR.FILESIZE_TOO_LARGE
         });
         fileErrors.push(fileError);
+        this.setState({fileErrors});
         continue;
       }
-
-      allFilePromises.push(this.readFile(file));
-    }
-
-    this.setState({
-      fileErrors
-    });
-
-    const {singleImage} = this.props;
-
-    Promise.all(allFilePromises).then(newFilesData => {
-      const dataURLs = singleImage?[]:this.state.pictures.slice();
-      const files = singleImage?[]:this.state.files.slice();
-
-      newFilesData.forEach(newFileData => {
-        this.uploadFileToServer(dataURLs, files, newFileData);
+      // Check for duplication
+      if (!this.props.images.includes(file.name) || filesName.includes(file.name)){
+        fileError = Object.assign(fileError, {
```

Comment:
I note that `Object.assign` is an old way to do `fileError = {fileError, ...{k:v}}`

Reply:
This was inherited from the original library. I reckon not changing it as long as it's compatible.